File: UnitTests/Priority.cs
using Calculator;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System;

namespace UnitTests
{
    [TestClass]
    public class Priority
    {
        [TestMethod]
        public void Plus()
        {
            Assert.AreEqual(1, Core.Priority('+'));
        }
        [TestMethod]
        public void Minus()
        {
            Assert.AreEqual(1, Core.Priority('-'));
        }
        [TestMethod]
        public void Times()
        {
            Assert.AreEqual(2, Core.Priority('*'));
        }
        [TestMethod]
        public void Divide()
        {
            Assert.AreEqual(2, Core.Priority('/'));
        }
        [TestMethod]
        public void Bracket()
        {
            Assert.AreEqual(0, Core.Priority('('));
        }
        [TestMethod]
        public void Unrecognised()
        {
            Assert.AreEqual(0, Core.Priority('%'));
        }
    }
}

File: Calculator VB/UnitTests/Priority.vb

Imports Calculator
Imports Microsoft.VisualStudio.TestTools.UnitTesting
Imports System

Namespace UnitTests

    <TestClass>
    Public Class Priority

        <TestMethod>
        Public Sub Plus()
            Assert.AreEqual(1, Core.Priority("+"c))
        End Sub

        <TestMethod>
        Public Sub Minus()
            Assert.AreEqual(1, Core.Priority("-"c))
        End Sub

        <TestMethod>
        Public Sub Times()
            Assert.AreEqual(2, Core.Priority("*"c))
        End Sub

        <TestMethod>
        Public Sub Divide()
            Assert.AreEqual(2, Core.Priority("/"c))
        End Sub

        <TestMethod>
        Public Sub Bracket()
            Assert.AreEqual(0, Core.Priority("("c))
        End Sub

        <TestMethod>
        Public Sub Unrecognised()
            Assert.AreEqual(0, Core.Priority("%"c))
        End Sub
    End Class
End Namespace

'=======================================================
'Service provided by Telerik (www.telerik.com)
'Conversion powered by Refactoring Essentials.
'Twitter: @telerik
'Facebook: facebook.com/telerik
'=======================================================
